sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3447,12 +3447,14 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
 
     val leftKeys = joinNames.map { keyName =>
       left.output.find(attr => resolver(attr.name, keyName)).getOrElse {
-        throw QueryCompilationErrors.unresolvedUsingColForJoinError(keyName, left, "left")
+        throw QueryCompilationErrors.unresolvedUsingColForJoinError(
+          keyName, left.schema.fieldNames.map(toSQLId).mkString(", "), "left")
```

Comment:
The `suggestion` list is not stable, could you sort it, please.

Reply:
Sorted!